mm0-rs/README.md

```diff
@@ -6,19 +6,80 @@ Watch this space for more updates, as `mm0-rs` is under active development. See
+Then cd into this `mm0-rs` directory (using `cd mm0/mm0-rs`) and run:
 
     cargo build --release
 
-from the `mm0-rs` directory, then copy or symlink the resulting executable file `target/release/mm0-rs` to your system path and/or point `vscode-mm0` to it using the setting
+You then should make it easy to execute.
+One way is to copy or symlink the resulting executable file `target/release/mm0-rs` to your system path.
+An alternative, if you're integrating with Visual Studio Code as suggested
+below, is to point `vscode-mm0` to the executable using (as
+explained below) the setting "metamath-zero.executablePath": "mm0-rs"
+in your `settings.json` file.
 
-    "metamath-zero.executablePath": "mm0-rs",
+## Integration with Visual Studio Code
 
-in your `settings.json` file.
+We typically use Visual Studio Code, an open source software
+editor / IDE which supports LSP servers.
+If you choose to do the same,
+[download and install Visual Studio Code](https://code.visualstudio.com/).
+
+Select View/Extensions, search for "Metamath"
+Look for "Metamath Zero" & press its "install" button.
+
+If `mm0-rs` is not on your path, then you need to
+change the Visual Studio Code settings so it can execute `mm0-rs`.
+To do this, after you've installed "Metamath Zero" look at the name,
+click on its small "gear symbol" to change its settings.
+Note: if you click on its "gear symbol" and see nothing, that's a bug
+in the Visual Studio Code program; restart it and again select View/Extensions
+to you can see the settings to change them.
+Now modify the "Metamath-zero Executable Path" setting (which corresponds
+to `metamath-zero.executablePath` in `settings.json`)
+to be the correct path for the executable
+(it defaults to `mm0-rs`).
+You should probably use an absolute path (e.g., one starting with "/"
+on Linux/Unix/MacOS).
 
 ## Usage
 
+The `mm0-rs` program provides a number of modes and options if you
+want to invoke it directly. For example:
+
 * `mm0-rs server` causes it to send and receive LSP server commands via stdin and stdout. This is not used directly from the CLI but rather is invoked by `vscode-mm0` when it is set up to use `mm0-rs` as a language server.
-  * `mm0-rs server --debug` is run by `vscode-mm0` when the extension itself is run in debugging mode, and this will enable backtraces and logging.
+* `mm0-rs server --debug` is run by `vscode-mm0` when the extension itself is run in debugging mode, and this will enable backtraces and logging.
+* `mm0-rs compile foo.mm1` will compile an MM1 file, reporting errors to the console. This is essentially the console version of the `server` mode.
+
+You can easily use `mm0-rs` from within Visual Studio Code.
+Start Visual Studio Code, then use File/Open,
+and navigate to the mm0/examples directory.
+Select a file to look at; `do-block.mm1` and `demo.mm1`
+are good starting points.
+Select "View/Problems" to see a summary.
+
+File `demo.mm1` show what proofs look like in MM1.
```

Comment:
This information should go with the MM1 language description; probably this file should link to https://github.com/digama0/mm0/blob/master/mm0-hs/mm1.md , which is being kept up to date even though `mm0-hs` itself isn't.

Reply:
Adding the cross-reference sounds great. I'd still like to put a quickstart here though.